### DISCUSSION TITLE
AXP202 Power Consumption

### DISCUSSION THREAD

**Discussion by SiYa2020** [micropython/micropython#9894] [Hardware & Peripherals]
Hello together, 
I hope you can help me.
I could not find in the data sheet,how much power the axp consumes. 
I would be very grateful if someone knows. 

Thanks a lot
Mira

**Comment by robert-hh** [comment]
Is it this kind of device: `Single Cell Li-Battery PWM Charger and Power System Management IC `
Data sheets are available.

**Reply by SiYa2020** [reply]
Hello robert-hh,

thank you for your reply.
I am new to the field and could not find or understand the power consumption of the AXP202 in the data sheet.
That is why I am looking for help here.

Thanks
[AXP202_PMU.pdf](https://github.com/micropython/micropython/files/9970982/AXP202_PMU.pdf)

**Reply by jimmo** [reply]
@SiYa2020 This is the MicroPython forum. Your question doesn't seem to have anything to do with MicroPython.

**Reply by SiYa2020** [reply]
Hello @jimmo,

thank you for your reply.
Sorry, I forgot to mention that I want to measure the current on the Lilygo T-Watch and the programs in Micropython.
I plan to measure all internal components for their power consumption.
That's why I wrote here in the forum.

Thanks a lot
Mira

**Reply by jimmo** [reply]
Power consumption of a component is much more complicated than a single number, that's why you can't just find a simple answer in the data sheet. For one thing it will be load-dependent and different based on whether it's charging and what the battery voltage is.

But if you actually want to _measure_ power consumption, then the data sheet isn't going to help you. You will actually need to measure it.

**Reply by SiYa2020** [reply]
Hi @jimmy,

thank you for your reply.
You have helped me a lot.

Now I know why there is nothing in the data sheet. 

Thank you very much!
Mira

**Reply by jimmo** [reply]
@SiYa2020 This is not what I'm saying at all! The data sheet is a gold mine of information -- it gives you efficiency curves and on-resistances etc. It's just not as simple as saying "this part uses this many milliwatts".

**Reply by SiYa2020** [reply]
Hi @jimmy,
I know the datasheet is a goldmine informatively. 

For me as a beginner it is not so easy but I struggle!

If I want to measure the power consumption....which registers can I address?

The T-Watch is powered by a lithium battery (380mAh, 3.7V) when not connected to USB.

**Reply by jimmo** [reply]
What are you actually trying to do?

Your initial question was "how much power the axp consumes." But now I think you might actually be asking how much power the axp is supplying to the rest of the watch. This is definitely all in the data sheet. There are registers for accessing the charge and discharge current as well as the coulomb counters. These registers are all described in the data sheet.

Suggest you look at https://github.com/tuupola/axp202 for an example.

**Reply by SiYa2020** [reply]
Hi @jimmy,

no, you have misunderstood me, sorry.

I want to measure how much the AXP consumes and not how much the AXP supplies the internal components with power.

Do I have to open the T-Watch to measure the current or is there another way?

**Reply by davefes** [reply]
To clarify:
- are you asking how much power is being delivered by the battery to the rest of the circuit OR
- are you asking what the self-consumption or operating current of the AXP is?

For the first case unless someone tells you how much either the watch circuitry consumes or can answer that question, you will have to do it yourself.
For the second case that specification should be in the datasheet.

**Reply by SiYa2020** [reply]
Hallo @davefes,

thank you for your help!

many thanks
